tests/config-ui/components/test-class-component-connect-gsc.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * @package WPSEO\UnitTests
+ */
+
+/**
+ * Class WPSEO_Config_Component_Connect_Google_Search_Console_Mock
+ */
+class WPSEO_Config_Component_Connect_Google_Search_Console_Mock extends WPSEO_Config_Component_Connect_Google_Search_Console {
+	protected $profile;
+
+	/** @var WPSEO_UnitTestCase */
+	protected $test;
+
+	public function __construct( $test ) {
+		$this->test = $test;
+
+		parent::__construct();
+	}
+
+	/**
+	 * Make function public
+	 *
+	 * @param array $current_data
+	 * @param array $data
+	 */
+	public function handle_access_token_clear( $current_data, $data ) {
+		return parent::handle_access_token_clear( $current_data, $data );
+	}
+
+	/**
+	 * Make function public
+	 *
+	 * @param array $current_data
+	 * @param array $data
+	 */
+	public function handle_profile_change( $current_data, $data ) {
+		return parent::handle_profile_change( $current_data, $data );
+	}
+
+	/**
+	 * Get the set service
+	 *
+	 * @return WPSEO_GSC_Service
+	 */
+	public function get_service() {
+		return $this->gsc_service;
+	}
+
+	/**
+	 * Get the current settings
+	 *
+	 * @return string
+	 */
+	public function get_settings() {
+		return $this->gsc_settings;
+	}
+
+	/**
+	 * Get mappings
+	 *
+	 * @return array
+	 */
+	public function get_mapping() {
+		return $this->mapping;
+	}
+
+	public function get_profile() {
+		return $this->profile;
+	}
+
+	public function set_profile( $profile ) {
+		$this->profile = $profile;
+	}
+
+	public function reload_issues() {
+		$this->test->stub_call_register( 'reload_issues' );
+	}
+
+	public function clear_gsc_data() {
+		$this->test->stub_call_register( 'clear_data' );
+	}
+}
+
+/**
+ * Class WPSEO_Config_Component_Connect_Google_Search_Console_Test
+ */
+class WPSEO_Config_Component_Connect_Google_Search_Console_Test extends WPSEO_UnitTestCase {
```

Comment:
I prefer using the built-in PHPUnit TestCase because it isn't coupled to the WordPress test suite and we want to move away from using that test suite.